conan/tools/microsoft/msbuilddeps.py

```diff
@@ -126,10 +127,13 @@ def _condition(self):
     @staticmethod
     def _dep_name(dep, build):
         dep_name = dep.ref.name
-        dep_name = dep_name.replace(".", "_")
         if build:  # dep.context == CONTEXT_BUILD:
             dep_name += "_build"
-        return dep_name
+        return MSBuildDeps._get_valid_xml_format(dep_name)
```

Comment:
```python
        return re.sub(dep_name, ....)
```
As a general rule I try to follow, I try to keep things as local as possible, no need to add new methods if they are only called from one place, and they are single line. Also I wouldn't add a file scope "constant" to be used just in one function, I would put it inside the function first. 

No need to change anything, it is good as-is, but I think in the long term is better to work that way.